federation/apis/federation/annotations.go

```diff
@@ -17,8 +17,9 @@ limitations under the License.
 package federation
 
 // FederationNameAnnotation is the annotation which holds the name of
-// the federation that an object is associated with. It must be
-// applied to all API objects associated with that federation.
+// the federation that that a federation control plane component is associated
+// with. It must be applied to all the API types that represent that federation
+// control plane components in the host cluster.
```

Comment:
@madhusudancs Currently, I've been annotating objects in both the host and joining clusters. Do we just want to annotate objects in the joining cluster?